### ISSUE TITLE
Add DXCompiler

### ISSUE THREAD

**Issue by DylanFPS** [sharpdx/SharpDX#1119]
[DXCompiler](https://github.com/Microsoft/DirectXShaderCompiler) is a (relatively) new library for compiling HLSL into DXIL, to then be used by DirectX. It can also compile HLSL into SPIR-V, which can then be compiled into a variety of other shading languages.

My proposal is that the SharpDX team introduces this package to the SharpDX stack, as there currently is no other C# bindings library for DXCompiler.

**Comment by amerkoleci** [comment]
I have dxc compiler in my game framework https://github.com/amerkoleci/vortice/blob/master/src/assets/Vortice.Assets/Graphics/ShaderCompiler.cs

**Comment by DylanFPS** [comment]
Oh awesome! Seeing as it's MIT, is it alright if I use it in my project? (having retained the copyright info of course)

**Comment by amerkoleci** [comment]
Of course, just keep the license head or clarify the usage, for me is fine :)

**Comment by DylanFPS** [comment]
Awesome, I'll take a few snippets but I'll probably end up porting all of the native interop to [AdvancedDLSupport](https://github.com/Firwood-Software/AdvanceDLSupport).
